bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java

```diff
@@ -3698,7 +3700,7 @@ public ServerConfiguration setHttpServerTrustStorePath(String trustStorePath) {
      * @return http server trustStore password
      */
     public String getHttpServerTrustStorePassword() {
-        return getString(HTTP_SERVER_KEY_STORE_PASSWORD);
+        return getString(HTTP_SERVER_TRUST_STORE_PASSWORD);
```

Comment:
An easier way to keep the compatibility and avoid introducing a new configuration:
```
return getString(HTTP_SERVER_TRUST_STORE_PASSWORD, getString(HTTP_SERVER_KEY_STORE_PASSWORD))
```

Reply:
We need to consider that the user configures the wrong trust store password, and then uses the correct key store password to successfully start the bookie. After the upgrade, the wrong trust store password takes effect, causing the bookie upgrade to fail?

This way like PR #3643, does not seem to be fully compatible.

Maybe we need more people to see this PR.